src/components/supervisor-details.tsx

```diff
@@ -0,0 +1,61 @@
+"use client";
+
+import { ArrowLeft } from "lucide-react";
+import Link from "next/link";
+import { notFound } from "next/navigation";
+
+import { useChats } from "@/hooks/use-chats";
+import { useSupervisors } from "@/hooks/use-supervisors";
+import { faculties } from "@/lib/faculties";
+import type { Supervisor } from "@/types/supervisor";
+
+function PaperEntry({ paper }: { paper: Supervisor["papers"][0] }) {
```

Comment:
hmm, can't we use `Paper` type here?

Reply:
We definitely can, I forgot that you made a separate interface for it. In case you ever wondered how you derive type of a single element from an array though, this is how 💀